src/record.rs

```diff
@@ -20,58 +21,149 @@ where
     /// ```
     pub fn new() -> Self {
         Self {
-            header: Default::default(),
+            ..Default::default()
         }
     }
 
     /// # Description
-    /// Returns number of bytes, struct `Record` uses on disk
-    pub fn size(&self) -> usize {
+    pub fn header_key_len(&self) -> u64 {
+        self.header.key_len
+    }
+
+    /// # Description
+    pub fn header_data_len(&self) -> u64 {
+        self.header.size - HEADER_UNALIGNED_SIZE as u64 - self.header.key_len
```

Comment:
may be we should store `data_size` as header size and key length should come from blob's header. Why waste space to store same value all the time in each record.

Reply:
Updated with ```bincode::serialized_size(_)```